pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -199,32 +214,195 @@ pub mod pallet {
+						&who,
+						amount,
+						Preservation::Expendable,
+					)?;
+					total_collateral_issuance.saturating_reduce(amount);
+				}
+
+				sum_of_issuances.saturating_reduce(burnt);
+
+				// if the total issuance drops to 0 due to this burn, kill the currency
+				if token_issuance <= &burnt {
+					dead_currencies.push(currency_id);
+				}
+			}
+
+			// destroy all active currencies if the collateral locked in the pool has dropped to 0
+			if total_collateral_issuance.is_zero() || sum_of_issuances.is_zero() {
+				// we assume all currencies that already had their total issuance at 0 have already been deactivated in a previous step
+				for (currency_id, _) in total_issuances {
+					T::Fungibles::start_destroy(currency_id, None)?;
+				}
+				Self::deposit_event(Event::RefundCompleted(pool_id.clone()));
+			} else {
+				// deactivate all currencies whose issuance dropped to 0 in this step
+				for currency_id in dead_currencies {
+					T::Fungibles::start_destroy(currency_id.clone(), None)?;
+				}
+			}
+
+			Ok(())
 		}
 
 		#[pallet::call_index(9)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn finish_destroy(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn finish_destroy(origin: OriginFor<T>, pool_id: T::PoolId) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let pool_details = Pools::<T>::get(pool_id.clone()).ok_or(Error::<T>::PoolUnknown)?;
+
+			ensure!(pool_details.state.is_destroying(), Error::<T>::InUse);
+
+			for currency_id in pool_details.bonded_currencies {
+				if T::Fungibles::asset_exists(currency_id.clone()) {
+					// This would fail with an InUse error if there are any accounts left on any currency
+					T::Fungibles::finish_destroy(currency_id.clone())?;
+				}
+			}
+
+			let pool_account = pool_id.clone().into();
+
+			let total_collateral_issuance = Self::get_pool_collateral(&pool_account);
+
+			if !total_collateral_issuance.is_zero() {
+				T::CollateralCurrency::transfer(
+					T::CollateralAssetId::get(),
+					&pool_account,
+					&pool_details.manager,
+					total_collateral_issuance,
+					Preservation::Expendable,
+				)?;
+			}
+
+			Pools::<T>::remove(&pool_id);
+
+			// TODO: refund deposit
+
+			Self::deposit_event(Event::Destroyed(pool_id));
+
+			Ok(())
 		}
 	}
 
-	impl<T: Config> Pallet<T> {}
+	impl<T: Config> Pallet<T> {
+		fn do_start_destroy_pool(pool_id: T::PoolId, maybe_check_owner: Option<AccountIdOf<T>>) -> DispatchResult {
+			let mut pool_details = Pools::<T>::get(pool_id.clone()).ok_or(Error::<T>::PoolUnknown)?;
+
+			ensure!(!pool_details.state.is_destroying(), Error::<T>::Destroying);
+
+			if let Some(caller) = maybe_check_owner {
+				ensure!(pool_details.is_manager(&caller), Error::<T>::Unauthorized);
+			}
+
+			pool_details.state.destroy();
+
+			Self::deposit_event(Event::DestructionStarted(pool_id.clone()));
+
+			let total_collateral_issuance = Self::get_pool_collateral(&pool_id.clone().into());
```

Comment:
The pool id type is configurable and can be anything as long as it implements `Into<Self::AccountId>`. Restricting it to be an account id doesn't hurt much but takes away some flexibility.

Reply:
Then I would simply keep the `Into<[u8; 32]>` bound, since accounts can always be created from such a type. Then you can do `AccountId::from(id.into())` to get your account ID. And the pool ID can now also be a simple string as well as long as it's ASCII and less than 32 characters long.

Reply:
also something that should be added to the wireframe as it's not related to destruction per se

Reply:
Also done in #772